component/portal/src/main/java/org/exoplatform/portal/mop/rest/NavigationRest.java

```diff
@@ -265,7 +276,45 @@ private List<ResultUserNode> convertNodes(Collection<UserNode> nodes) {
         continue;
       }
       ResultUserNode resultNode = new ResultUserNode(userNode);
-      resultNode.setChildren(convertNodes(userNode.getChildren()));
+      if (expandPageDetails && userNode.getPageRef() != null) {
+        Page userNodePage = layoutService.getPage(userNode.getPageRef());
+        boolean canEdit = false;
+        if (userNodePage.getEditPermission() != null && !userNodePage.getEditPermission().isBlank()){
+          canEdit = identity.isMemberOf(userNodePage.getEditPermission().split(":")[1],
+                  userNodePage.getEditPermission().split(":")[0]);
+        }
+        resultNode.setCanEditPage(canEdit);
+        HashMap<String, Object> editPermission = new LinkedHashMap<>();
+        editPermission.put("membershipType", userNodePage.getEditPermission().split(":")[0]);
+        try {
+          editPermission.put("group",
+                             organizationService.getGroupHandler().findGroupById(userNodePage.getEditPermission().split(":")[1]));
+        } catch (Exception e) {
+          editPermission.put("group", userNodePage.getEditPermission().split(":")[1]);
+        }
+        resultNode.setPageEditPermission(editPermission);
+        List<HashMap<String, Object>> accessPermissions = new ArrayList<>();
+
+        if(userNodePage.getAccessPermissions() != null && userNodePage.getAccessPermissions().length == 1 && userNodePage.getAccessPermissions()[0].equals("Everyone") ) {
```

Comment:
`userNodePage.getAccessPermissions()` != null

this should be a global condition for the two cases